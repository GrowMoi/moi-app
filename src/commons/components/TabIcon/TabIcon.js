import React, { Component } from 'react';
import { View, Text } from 'react-native';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import MoIcon from '../MoIcon/MoIcon';

const TabIconContainer = styled(View)`
  justify-content: center;
  align-items: center;
`;

const Title = styled(Text)`
  margin-top: 5;
  font-size: ${props => props.titleSize};
`;

export default class TabIcon extends Component {
  render() {
    const { name, size, title, selected, titleSize } = this.props;

    return (
      <TabIconContainer>
        <MoIcon name={name} size={size} active={selected} />
        <Title titleSize={titleSize} >{title}</Title>
      </TabIconContainer>
    );
  }
}

TabIcon.defaultProps = {
  size: 40,
  titleSize: 11,
  type: 'moi',
};

TabIcon.propTypes = {
  name: PropTypes.string,
  size: PropTypes.number,
  title: PropTypes.string,
  selected: PropTypes.bool,
  titleSize: PropTypes.number,
  type: PropTypes.string,
};
