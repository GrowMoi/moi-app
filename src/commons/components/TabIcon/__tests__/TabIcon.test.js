import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components/native';
import TabIcon from '../TabIcon';

describe('Moicon', () => {
  it('it should render correctly with default props', () => {
    const wrapper = shallow(<TabIcon />);
    const tree = wrapper.dive();

    expect(tree).toMatchSnapshot();
    expect(tree).toBeTruthy();
  });

  it('it should render without name prop', () => {
    const wrapper = renderer.create(<TabIcon name="task" />);
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
