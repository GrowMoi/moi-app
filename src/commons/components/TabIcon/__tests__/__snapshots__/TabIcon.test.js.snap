// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moicon it should render correctly with default props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <MoIcon
    badgeValue={0}
    fadeDuration={0}
    name="task"
    resizeMode="contain"
    size={40}
  />
  <Styled(Text)
    titleSize={11}
  />
</View>
`;

exports[`Moicon it should render without name prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <Image
      fadeDuration={0}
      resizeMode="contain"
      size={40}
      source={1}
      style={
        Array [
          Object {
            "height": 40,
            "width": 40,
          },
          undefined,
        ]
      }
      type="square"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 11,
          "marginTop": 5,
        },
        undefined,
      ]
    }
    titleSize={11}
  />
</View>
`;
